### ISSUE TITLE
Patcher tries to incorrectly apply patch if root element type is changed

### ISSUE THREAD

**Issue by Alikont** [AvaloniaCommunity/Avalonia.FuncUI#50]
If you change the root view type (e.g. from Button to TextBlock), the patcher will try to apply attributes to existing element (of the wrong type). 

This will cause exception during view update.

```
System.ArgumentException: Property 'Text not registered on 'Avalonia.Controls.Button
```

[Code](https://github.com/AvaloniaCommunity/Avalonia.FuncUI/blob/8b19f855424239a3bc463a725c5301c66f52e103/src/Avalonia.FuncUI/VirtualDom/VirtualDom.Patcher.fs#L220)

**Comment by JaggerJo** [comment]
Could you give me some additional information on how this error occurred ?

I'm able to reproduce it but I'm interested if you are using HostWindow / HostControl ?

**Comment by Alikont** [comment]
I’m using default AvaloniaUI template with Host Window

The only modifications I’ve done are:


  1.  Pass HostWindow to Update function (because I need to open OpenFileDialog)
  2.  Add withSyncDispatch to force updates on dispatcher thread (because I have a lot of async subscriptions and it caused invalid thread access)

I want for my app to have several screens, and I did switch between them in pattern match in View function.

My first view is “Welcome” screen that works like a loading splash screen and the second view  is “Working” view.

The first one was simple TextBlock, the second was Grid. When I started the app I saw weird exceptions, starting from NullReferenceException. I started to simplify my app bit by bit until I got one view as a TextBlock and other view as a Button, then I saw an exception about trying to apply a property that doesn’t exist, and that made sense. Then I went to FuncUI source code. Found the error.

After that I wrapped my entire view into a Border control and everything started to work. (Except for few additional things regarding ListBox and TextBox updates, that I will track in better details and confirm that it’s FuncUI issue, and not Avalonia issue).
